### ISSUE TITLE
XML / perlmod fortran output, the type of a parameter of a function is set to the name of the parameter

### ISSUE THREAD

**Issue by Chabrier** [doxygen/doxygen#9399]
**Describe the bug**
When using the perlmod mode on fortran code, i do not get the type of the parameter of a function.

**Expected behavior**
I would expect to get the type like with c code.

**Screenshots**
part of the code

```
function area_of_circle (r)  
  implicit none
  
  real :: area_of_circle  
  real :: r
  real :: pi

  pi = 4 * atan (1.0)

  area_of_circle = pi * r**2
  
end function area_of_circle

```
part of the output (DoxyDocs.pm)

```
 members => [
          {
            kind => 'function',
            name => 'area_of_circle',
            virtualness => 'non_virtual',
            protection => 'public',
            static => 'no',
            brief => {},
            detailed => {},
            type => 'real function',
            const => 'no',
            volatile => 'no',
            parameters => [
              {
                definition_name => 'r',
                type => 'r'                <--------------------------real expected
              }
            ]

```

**To Reproduce**

[toReproduce.zip](https://github.com/doxygen/doxygen/files/8899804/toReproduce.zip)

**Version**

$ ~/DEVS/doxygen/build/bin/doxygen --version
1.9.5 (886349348ad5dc60152e0d5b92e6ffbaf5f4a921*)
$ lsb_release -a
LSB Version:	core-11.1.0ubuntu2-noarch:printing-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal

**Additional context**

If this is something to fix, with some tracks, maybe i could contribute.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9400.

**Comment by Chabrier** [comment]
Very efficient patch in my case! thanks!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
